Document how to upgrade NVIDIA drivers manually (#216)
anbox-cloud/discourse-docs: howto/update/upgrade-anbox.md

    juju exec --all -- /bin/sh -c 'sudo apt update && sudo apt upgrade -y'

> If the LXD charm is deployed on a machine that has a NVIDIA GPU installed, running the above command for the machine may upgrade the NVIDIA drivers, which accidentally suspends running containers with GPU support. Starting with the 1.17.1 release, the NVIDIA drivers are held from being upgraded until you upgrade the LXD charm by using the Juju command. To check the current installed NVIDIA drivers are held from being upgraded:

Reviewer: ```suggestion
If the LXD charm is deployed on a machine that has an NVIDIA GPU installed, running the above command for the machine may upgrade the NVIDIA drivers, which accidentally suspends running containers with GPU support. Starting with the 1.17.1 release, the NVIDIA drivers are held from being upgraded until you upgrade the LXD charm by using the Juju command. To check if the current installed NVIDIA drivers are held from being upgraded:
```

Reviewer: Although NVIDIA starts with 'N', the starting sound is that of an vowel ('e'). So its 'an NVIDIA'.

> If the LXD charm is deployed on a machine that has an NVIDIA GPU installed, running the above command for the machine may upgrade the NVIDIA drivers, which accidentally suspends running containers with GPU support. Starting with the 1.17.1 release, the NVIDIA drivers are held from being upgraded until you upgrade the LXD charm by using the Juju command. To check if the current installed NVIDIA drivers are held from being upgraded:

Reviewer: ```suggestion
If the LXD charm is deployed on a machine that has an NVIDIA GPU installed, running the above command for the machine may upgrade the NVIDIA drivers, which accidentally suspends running containers with GPU support. Starting with the 1.17.1 release, the NVIDIA drivers are held from being upgraded until you upgrade the LXD charm using the Juju command. To check if the currently installed NVIDIA drivers are held from being upgraded:
```

>     apt-mark showhold libnvidia-.*-$(nvidia-smi --query-gpu=driver_version --format=csv,noheader | cut -d'.' -f1)

Reviewer 2: To avoid globbing (just in case):

```suggestion
    apt-mark showhold "libnvidia-.*-$(nvidia-smi --query-gpu=driver_version --format=csv,noheader | cut -d'.' -f1)"
```

If they are not, run the following command to do so:

>     sudo apt-mark hold linux-modules-nvidia-* nvidia-* libnvidia-*

Reviewer 2: ```suggestion
    sudo apt-mark hold 'linux-modules-nvidia-*' 'nvidia-*' 'libnvidia-*'
```

    juju run --wait=1m lxd/0 clear-notification

> If the LXD charm is deployed on a machine with a NVIDIA GPU installed, the NVIDIA drivers are held from being upgraded by default in case of downtime for all running containers due to either a manual upgrade or an [unattended-upgrade](https://wiki.debian.org/UnattendedUpgrades). The downside to this is that the machine may miss security updates for the NVIDIA drivers. To manually upgrade the NVIDIA drivers, you need to:

Reviewer: ```suggestion
If the LXD charm is deployed on a machine with an NVIDIA GPU installed, by default, the NVIDIA drivers are held from being upgraded in case of downtime for all running containers due to either a manual upgrade or an [unattended-upgrade](https://wiki.debian.org/UnattendedUpgrades). The downside to this is that the machine may miss security updates for the NVIDIA drivers. To manually upgrade the NVIDIA drivers, you need to:
```

> 1. Stop the all running containers

Reviewer: ```suggestion
1. Stop all running containers.
```

2. Unhold the NVIDIA drivers to allow an upgrade:

>         sudo apt-mark unhold linux-modules-nvidia-* nvidia-* libnvidia-*

Reviewer 2: ```suggestion
        sudo apt-mark unhold 'linux-modules-nvidia-*' 'nvidia-*' 'libnvidia-*'
```

Author: We have a task scheduled in this pulse.
Yes, it would be there.
Let's keep it as is.

3. Update OS:

        sudo apt update && sudo apt upgrade -y

4. Hold the NVIDIA drivers from being updated again:

>         sudo apt-mark hold linux-modules-nvidia-* nvidia-* libnvidia-*

Reviewer 2: ```suggestion
        sudo apt-mark hold 'linux-modules-nvidia-*' 'nvidia-*' 'libnvidia-*'
```

> A Juju action that simplifies the manual upgrade for the NVIDIA drivers will be added in the future release.

Reviewer: ```suggestion
A Juju action that simplifies the manual upgrade for NVIDIA drivers will be added in a future release.
```

Reviewer 2: Isn't it risky to write 'will be added' here? Shouldn't we be more careful with the wording? I think it would be wise to be more prudent:
```suggestion
A Juju action that simplifies the manual upgrade for the NVIDIA drivers may be added in the future release.
```

Reviewer: It is not uncommon to have such statements that talk about future releases but we must be sure that it is a fix that is absolutely going to be implemented - something without which we wouldn't go further. If it is not that certain, then we shouldn't include it.
